File: app/src/main/java/taiwan/no1/app/ssfm/customized/FontTextView.java
package taiwan.no1.app.ssfm.customized;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

/**
 * TODO: document your custom view class.
 */
public class FontTextView extends View {
    public FontTextView(Context context) {
        super(context);
        init(null, 0);
    }

    public FontTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs, 0);
    }

    public FontTextView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(attrs, defStyle);
    }

    private void init(AttributeSet attrs, int defStyle) {
        // Load attributes
    }
}

File: app/src/main/kotlin/taiwan/no1/app/ssfm/customized/FontTextView.kt
package taiwan.no1.app.ssfm.customized

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import android.widget.TextView
import taiwan.no1.app.ssfm.R

/**
 * For inputting customize special font.
 */
class FontTextView: TextView {
    constructor(context: Context): super(context) {
        this.init(null, 0)
    }

    constructor(context: Context, attrs: AttributeSet): super(context, attrs) {
        this.init(attrs, 0)
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int): super(context, attrs, defStyle) {
        this.init(attrs, defStyle)
    }

    private fun init(attrs: AttributeSet?, defStyle: Int) {
        context.obtainStyledAttributes(attrs, R.styleable.FontTextView).also {
            it.getString(R.styleable.FontTextView_font)?.let {
                this.typeface = Typeface.createFromAsset(this.context.assets, "fonts/$it")
            }
        }.recycle()
    }
}
